plugins/route/allocation/simple/plugin.go

```diff
@@ -0,0 +1,72 @@
+package simple
+
+import (
+	"errors"
+	"fmt"
+
+	"code.google.com/p/go-uuid/uuid"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/golang/glog"
+
+	routeapi "github.com/openshift/origin/pkg/route/api"
+)
+
+// Default DNS suffix to use if no configuration is passed to this plugin.
+// Would be better if we could use "v3.openshift.app", someone bought that!
+const defaultDNSSuffix = "v3.openshift.com"
+
+// SimpleAllocationPlugin implements the route.AllocationPlugin interface
+// to provide a simple unsharded (or single sharded) allocation plugin.
+type SimpleAllocationPlugin struct {
+	DNSSuffix string
+}
+
+// Creates a new SimpleAllocationPlugin.
+func NewSimpleAllocationPlugin(suffix string) (*SimpleAllocationPlugin, error) {
+	if len(suffix) == 0 {
+		suffix = defaultDNSSuffix
+	}
+
+	glog.V(4).Infof("NewSimpleAllocationPlugin: suffix=%s", suffix)
+
+	// Check that the DNS suffix is valid.
+	if !util.IsDNSSubdomain(suffix) {
+		errmsg := fmt.Sprintf("invalid DNS suffix: %s", suffix)
+		glog.Errorf("NewSimpleAllocationPlugin: %s", errmsg)
```

Comment:
Generally if you return an error don't bother logging - collapse to fmt.Errorf() which is more idiomatic